File: Quivi.Backoffice.Api/Requests/Reporting/GetProductSalesRequest.cs
using Quivi.Infrastructure.Abstractions.Repositories.Criterias;

namespace Quivi.Backoffice.Api.Requests.Reporting
{
    public class GetProductSalesRequest : APagedRequest
    {
        public bool? AdminView { get; init; }
        public SalesPeriod? Period { get; init; }
        public ProductSalesSortBy SortBy { get; init; }
        public DateTimeOffset? From { get; init; }
        public DateTimeOffset? To { get; init; }
    }
}

File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/reporting/GetProductSalesRequest.ts
import { PagedRequest } from "../PagedRequest";
import { ProductSalesSortBy } from "./ProductSalesSortBy";
import { SalesPeriod } from "./SalesPeriod";

export interface GetProductSalesRequest extends PagedRequest {
    readonly adminView?: boolean;
    readonly sortBy: ProductSalesSortBy;
    readonly period?: SalesPeriod;
    readonly from?: string;
    readonly to?: string;
}